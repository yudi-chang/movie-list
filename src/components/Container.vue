<template>
  <div class="container">
    <slot></slot>
  </div>
</template>

<style scoped lang="scss">
.container {
  width: 1250px;
  max-width: 100%;
  margin: 0 auto;
  padding: 20px;
  transition: width 0.3s ease-in-out;
}

@media (max-width: $breakpoint-lg) {
  .container {
    width: 100%;
    padding: 15px;
  }
}

@media (max-width: $breakpoint-md) {
  .container {
    padding: 10px;
  }
}
</style>
