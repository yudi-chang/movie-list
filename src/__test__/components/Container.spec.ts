import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import Container from '@/components/Container.vue'

describe('Container', () => {
  const defaultParams = {}

  const createWrapper = (params = defaultParams) => {
    return mount(Container, params)
  }

  it('should renders correctly', async () => {
    // no logic in this component, so just use matchSnapshot
    const wrapper = createWrapper();
    expect(wrapper.element).toMatchSnapshot();
  })
})
