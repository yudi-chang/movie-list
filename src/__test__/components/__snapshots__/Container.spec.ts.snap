// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > should renders correctly 1`] = `
<div
  class="container"
  data-v-85d507ef=""
>
  
  
</div>
`;
